<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2024-12-27" />

<title>reproduce figure1 a b c</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">FigRemake with Chatomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/crazyhottommy/reproduce_genomics_paper_figures">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">reproduce figure1 a b c</h1>
<h4 class="date">2024-12-27</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-01-11
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>reproduce_genomics_paper_figures/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20241226code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20241226)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20241226code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20241226)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomcrazyhottommyreproducegenomicspaperfigurestree44ade109d19b18d14b3ce69f08cd0e5c4e6bdd13targetblank44ade10a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/crazyhottommy/reproduce_genomics_paper_figures/tree/44ade109d19b18d14b3ce69f08cd0e5c4e6bdd13" target="_blank">44ade10</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomcrazyhottommyreproducegenomicspaperfigurestree44ade109d19b18d14b3ce69f08cd0e5c4e6bdd13targetblank44ade10a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/crazyhottommy/reproduce_genomics_paper_figures/tree/44ade109d19b18d14b3ce69f08cd0e5c4e6bdd13" target="_blank">44ade10</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/figure/
    Ignored:    data/fastq/
    Ignored:    data/public_data/
    Ignored:    data/reference/

Unstaged changes:
    Modified:   .gitignore
    Modified:   analysis/04_figure1_a_b_c.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/04_figure1_a_b_c.Rmd</code>) and
HTML (<code>docs/04_figure1_a_b_c.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/crazyhottommy/reproduce_genomics_paper_figures/blob/a1bdd913fb482f7417b3ebbc22bd0669712a6b4c/analysis/04_figure1_a_b_c.Rmd" target="_blank">a1bdd91</a>
</td>
<td>
crazyhottommy
</td>
<td>
2025-01-04
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/crazyhottommy/reproduce_genomics_paper_figures/a1bdd913fb482f7417b3ebbc22bd0669712a6b4c/docs/04_figure1_a_b_c.html" target="_blank">a1bdd91</a>
</td>
<td>
crazyhottommy
</td>
<td>
2025-01-04
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/crazyhottommy/reproduce_genomics_paper_figures/blob/c9a4ca2d11835c56a2b7685b05e4a683ab50bbed/analysis/04_figure1_a_b_c.Rmd" target="_blank">c9a4ca2</a>
</td>
<td>
crazyhottommy
</td>
<td>
2024-12-31
</td>
<td>
version 0.1
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/crazyhottommy/reproduce_genomics_paper_figures/c9a4ca2d11835c56a2b7685b05e4a683ab50bbed/docs/04_figure1_a_b_c.html" target="_blank">c9a4ca2</a>
</td>
<td>
crazyhottommy
</td>
<td>
2024-12-31
</td>
<td>
version 0.1
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Now the fun starts as we finished all the pre-processing steps. Let’s
recreate the figure 1a,b,c!</p>
<p><img src="imgs/figabc.png" /> ### Figure 1a</p>
<p>Figure 1a and 1b are venn-diagram showing the overlapping peak number
of YAP/TAZ and YAP/TAZ/TEAD4</p>
<p>To find overlap/intersection of genomic regions, one can use the
famous <a
href="https://bedtools.readthedocs.io/en/latest/index.html"><code>bedtools</code></a>
on command line or use <code>GenomicRanges</code> in <code>R</code>.</p>
<div id="lets-try-bedtools-first." class="section level4">
<h4>Let’s try bedtools first.</h4>
<pre class="bash"><code>conda install bedtools</code></pre>
<p>The most frequently used command is the <a
href="https://bedtools.readthedocs.io/en/latest/content/tools/intersect.html"><code>bedtools intersect</code></a>
command.</p>
<p><img src="imgs/bedtools.png" /></p>
<pre class="bash"><code>cd data/fatstq

bedtools intersect -a TAZ_peak/TAZ_peaks.narrowPeak -b YAP_peak/YAP_peaks.narrowPeak -wa | wc -l

7199
</code></pre>
<p>There are <code>7199</code> TAZ peaks overlap with YAP1 peaks.</p>
<p>I usually pipe the output to sort | uniq to get the unique TAZ peaks
because there are could be one TAZ peak overlapping with mulitiple YAP1
peaks and it gets repeated in the <code>bedtools</code> output.</p>
<pre class="bash"><code>bedtools intersect -a TAZ_peak/TAZ_peaks.narrowPeak -b YAP_peak/YAP_peaks.narrowPeak -wa | sort | uniq |  wc -l

7154</code></pre>
<p>There are <code>7154</code> <strong>unique</strong> TAZ peaks overlap
with YAP1 peaks.</p>
</div>
<div id="use-genomicranges-in-r" class="section level4">
<h4>Use GenomicRanges in R</h4>
<p>Let’s use R instead with the <code>GenomicRanges</code> package. Read
my <a
href="https://divingintogeneticsandgenomics.com/post/genomic-interval/">blog
post</a> on why we need to learn how to use it.</p>
<pre class="r"><code>library(GenomicRanges)
library(rtracklayer) # for reading in bed file
library(here)
TAZ_peaks&lt;- import(here(&quot;data/fastq/TAZ_peak/TAZ_peaks.narrowPeak&quot;))
YAP_peaks&lt;- import(here(&quot;data/fastq/YAP_peak/YAP_peaks.narrowPeak&quot;))

TAZ_peaks</code></pre>
<pre><code>#&gt; GRanges object with 10719 ranges and 6 metadata columns:
#&gt;           seqnames              ranges strand |           name     score
#&gt;              &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;character&gt; &lt;numeric&gt;
#&gt;       [1]     chr1     1024631-1025009      * |     TAZ_peak_1       496
#&gt;       [2]     chr1     1079586-1079774      * |     TAZ_peak_2        35
#&gt;       [3]     chr1     1264883-1265653      * |     TAZ_peak_3       244
#&gt;       [4]     chr1     1360608-1360960      * |     TAZ_peak_4       349
#&gt;       [5]     chr1     1659316-1659509      * |     TAZ_peak_5        54
#&gt;       ...      ...                 ...    ... .            ...       ...
#&gt;   [10715]     chrX 154596526-154596874      * | TAZ_peak_10715       282
#&gt;   [10716]     chrX 154600477-154600851      * | TAZ_peak_10716       135
#&gt;   [10717]     chrX 154604091-154604381      * | TAZ_peak_10717        64
#&gt;   [10718]     chrX 154732674-154733351      * | TAZ_peak_10718       103
#&gt;   [10719]     chrX 155888192-155888475      * | TAZ_peak_10719       136
#&gt;           signalValue    pValue    qValue      peak
#&gt;             &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;integer&gt;
#&gt;       [1]    15.48650  54.10540  49.66990       192
#&gt;       [2]     4.00512   6.64940   3.52515        11
#&gt;       [3]    11.36980  28.52280  24.49830       167
#&gt;       [4]    15.08250  39.17130  34.93810       174
#&gt;       [5]     5.41424   8.77757   5.48809        50
#&gt;       ...         ...       ...       ...       ...
#&gt;   [10715]    13.14890  32.33060  28.22330       200
#&gt;   [10716]     8.50809  17.30320  13.59470       187
#&gt;   [10717]     5.80097   9.75655   6.40776       170
#&gt;   [10718]     7.34790  13.93350  10.35880       496
#&gt;   [10719]     7.20922  17.34310  13.62840       155
#&gt;   -------
#&gt;   seqinfo: 28 sequences from an unspecified genome; no seqlengths</code></pre>
<pre class="r"><code>YAP_peaks</code></pre>
<pre><code>#&gt; GRanges object with 9807 ranges and 6 metadata columns:
#&gt;          seqnames              ranges strand |          name     score
#&gt;             &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |   &lt;character&gt; &lt;numeric&gt;
#&gt;      [1]     chr1          9894-10324      * |    YAP_peak_1        81
#&gt;      [2]     chr1       180610-180805      * |    YAP_peak_2        44
#&gt;      [3]     chr1     1024628-1025059      * |    YAP_peak_3       494
#&gt;      [4]     chr1     1264837-1265155      * |    YAP_peak_4       148
#&gt;      [5]     chr1     1265320-1265695      * |    YAP_peak_5       131
#&gt;      ...      ...                 ...    ... .           ...       ...
#&gt;   [9803]     chrX 154600351-154600918      * | YAP_peak_9803       131
#&gt;   [9804]     chrX 154732680-154732891      * | YAP_peak_9804        58
#&gt;   [9805]     chrX 154797063-154797337      * | YAP_peak_9805        49
#&gt;   [9806]     chrX 155888248-155888487      * | YAP_peak_9806       108
#&gt;   [9807]     chrX 156030280-156030812      * | YAP_peak_9807       370
#&gt;          signalValue    pValue    qValue      peak
#&gt;            &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;integer&gt;
#&gt;      [1]     2.30543  11.76320   8.16671       159
#&gt;      [2]     2.45356   7.78457   4.48630       172
#&gt;      [3]    16.37320  54.38830  49.46070       192
#&gt;      [4]     8.81841  18.84720  14.89750       188
#&gt;      [5]     8.62413  17.00830  13.14010       112
#&gt;      ...         ...       ...       ...       ...
#&gt;   [9803]     8.62413  17.00830  13.14010       417
#&gt;   [9804]     5.74942   9.32421   5.89513        78
#&gt;   [9805]     5.29420   8.25988   4.92389       140
#&gt;   [9806]     6.72473  14.58900  10.84320       151
#&gt;   [9807]     2.80955  41.71780  37.07200       273
#&gt;   -------
#&gt;   seqinfo: 27 sequences from an unspecified genome; no seqlengths</code></pre>
<p>We have 9807 YAP1 peaks and 10719 TAZ peaks. How many of them
overlap?</p>
<pre class="r"><code>TAZ_overlap_YAP_peaks&lt;- subsetByOverlaps(TAZ_peaks, YAP_peaks)
length(TAZ_overlap_YAP_peaks)</code></pre>
<pre><code>#&gt; [1] 7154</code></pre>
<pre class="r"><code>YAP_overlap_TAZ_peaks&lt;- subsetByOverlaps(YAP_peaks, TAZ_peaks)
length(YAP_overlap_TAZ_peaks)</code></pre>
<pre><code>#&gt; [1] 7164</code></pre>
<p>so we have 7154 out of 10719 TAZ peaks overlapping with YAP1 peaks.
(Note, this is the same number we got from
<code>bedtools intersect</code>).</p>
<p>and we have 7164 out of 9807 YAP1 peaks overlapping with TAZ
peaks.</p>
<p>The venn-diagram needs a common number in the intersection. How do we
deal with it? There are different decisions you can make and it does not
affect the conclusion of the figure: most of the TAZ and YAP1 peaks
overlap!</p>
<p>We can just use the number of YAP1 peaks that overlap with TAZ as the
intersection.</p>
<p>There are many packages to make a venndiagram. I use <a
href="https://github.com/js229/Vennerable"><code>Vennerable</code></a>.</p>
<pre class="r"><code># devtools::install_github(&quot;js229/Vennerable&quot;)

library(Vennerable)

n_YAP &lt;- length(YAP_peaks)  # Total peaks 
n_TAZ &lt;- length(TAZ_peaks)  # Total peaks 

n_overlap &lt;- length(YAP_overlap_TAZ_peaks)

venn_data &lt;- Venn(SetNames = c(&quot;YAP&quot;, &quot;TAZ&quot;),
                  Weight = c(
                    &quot;10&quot; = n_YAP, # Unique to A
                    &quot;01&quot; = n_TAZ, # Unique to B
                    &quot;11&quot; = n_overlap         # Intersection
                  ))

# Plot the Venn diagram
plot(venn_data)</code></pre>
<p><img src="figure/04_figure1_a_b_c.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/crazyhottommy/reproduce_genomics_paper_figures/blob/c9a4ca2d11835c56a2b7685b05e4a683ab50bbed/docs/figure/04_figure1_a_b_c.Rmd/unnamed-chunk-7-1.png" target="_blank">c9a4ca2</a>
</td>
<td>
crazyhottommy
</td>
<td>
2024-12-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>You can also use <code>makeVennDiagram</code> in the
<code>ChIPpeakAnno</code> package.</p>
<p>In our case, we already have the number of the two sets and the
intersection, so we used <code>vennerable</code>.</p>
<p>Take a look at <a
href="https://github.com/gaospecial/ggVennDiagram">ggVennDiagram</a>
too</p>
</div>
<div id="figure-1b" class="section level3">
<h3>Figure 1b</h3>
<p>We can easily make Figure 1b now that we have some foundations.</p>
<pre class="r"><code>TEAD4_peak&lt;- import(here(&quot;data/fastq/TEAD4_peak/TEAD4_peaks.narrowPeak&quot;))

YAP_overlap_TAZ_peaks_overlap_TEAD4&lt;- subsetByOverlaps(YAP_overlap_TAZ_peaks, TEAD4_peak)

n_YAP_TAZ &lt;- length(YAP_overlap_TAZ_peaks)  # Total peaks 
n_TEAD4 &lt;- length(TEAD4_peak)  # Total peaks 
n_overlap2&lt;- length(YAP_overlap_TAZ_peaks_overlap_TEAD4)

venn_data2 &lt;- Venn(SetNames = c(&quot;YAP/TAZ&quot;, &quot;TEAD4&quot;),
                  Weight = c(
                    &quot;10&quot; = n_YAP_TAZ, # Unique to A
                    &quot;01&quot; = n_TEAD4, # Unique to B
                    &quot;11&quot; = n_overlap2        # Intersection
                  ))

# Plot the Venn diagram
plot(venn_data2)</code></pre>
<p><img src="figure/04_figure1_a_b_c.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/crazyhottommy/reproduce_genomics_paper_figures/blob/c9a4ca2d11835c56a2b7685b05e4a683ab50bbed/docs/figure/04_figure1_a_b_c.Rmd/unnamed-chunk-8-1.png" target="_blank">c9a4ca2</a>
</td>
<td>
crazyhottommy
</td>
<td>
2024-12-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="figure-1c" class="section level3">
<h3>Figure 1c</h3>
<p>This figure requires a little more work. Let’s decompose it.</p>
<p>Description of the figure in the paper:</p>
<blockquote>
<ol start="3" style="list-style-type: lower-alpha">
<li>Position of TEAD4 peak summits relative to the summits of the
overlapping YAP/TAZ peaks, in a 500 bp window surrounding the summit of
YAP/TAZ peaks.</li>
</ol>
</blockquote>
<blockquote>
<p>TAZ peaks coordinates and summit positions were used to represent
common peaks between YAP and TAZ peaks (YAP/TAZ peaks) and were used
when comparing YAP/TAZ peaks with other ChIP-seq data.</p>
</blockquote>
<ul>
<li>data we need: the TAZ peak set and the TEAD4 peak set.</li>
<li>x-axis: when a TEAD peak overlaps with a TAZ peak, the distance
between the summit of the the TAZ peak and the TEAD4 peak summit.</li>
<li>y-axis: the number of TEAD peaks for each distance</li>
</ul>
<p>A summit is the highest signal point within a peak.
<code>MACS3</code> outputs that.</p>
<pre class="r"><code>TAZ_summit&lt;- import(here(&quot;data/fastq/TAZ_peak/TAZ_summits.bed&quot;))

TAZ_summit&lt;- TAZ_summit[TAZ_summit$name %in% TAZ_overlap_YAP_peaks$name]

TEAD4_summit&lt;- import(here(&quot;data/fastq/TEAD4_peak/TEAD4_summits.bed&quot;))

TEAD4_summit</code></pre>
<pre><code>#&gt; GRanges object with 11512 ranges and 2 metadata columns:
#&gt;           seqnames    ranges strand |             name     score
#&gt;              &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |      &lt;character&gt; &lt;numeric&gt;
#&gt;       [1]     chr1   1024830      * |     TEAD4_peak_1  37.76480
#&gt;       [2]     chr1   1079611      * |     TEAD4_peak_2   5.27663
#&gt;       [3]     chr1   1265037      * |     TEAD4_peak_3  45.50160
#&gt;       [4]     chr1   1360763      * |     TEAD4_peak_4  63.99220
#&gt;       [5]     chr1   1659345      * |     TEAD4_peak_5   6.62232
#&gt;       ...      ...       ...    ... .              ...       ...
#&gt;   [11508]     chrX 154733092      * | TEAD4_peak_11508   8.50667
#&gt;   [11509]     chrX 154735953      * | TEAD4_peak_11509  12.54790
#&gt;   [11510]     chrX 154967772      * | TEAD4_peak_11510   7.62224
#&gt;   [11511]     chrX 155057497      * | TEAD4_peak_11511   5.24858
#&gt;   [11512]     chrX 155888353      * | TEAD4_peak_11512   9.30619
#&gt;   -------
#&gt;   seqinfo: 24 sequences from an unspecified genome; no seqlengths</code></pre>
<p>They both represent a single base point that has the highest signal
in the peaks.</p>
<p>expand the TAZ summit to a 500bp window</p>
<pre class="r"><code>TAZ_500bp_window&lt;- resize(TAZ_summit, width = 500, fix=&quot;center&quot;)

hits&lt;- findOverlaps(TEAD4_summit, TAZ_500bp_window)

# a hits object with the indices of the overlapping query and subject
hits</code></pre>
<pre><code>#&gt; Hits object with 5901 hits and 0 metadata columns:
#&gt;          queryHits subjectHits
#&gt;          &lt;integer&gt;   &lt;integer&gt;
#&gt;      [1]         1           1
#&gt;      [2]         3           2
#&gt;      [3]         4           3
#&gt;      [4]         5           4
#&gt;      [5]         7           6
#&gt;      ...       ...         ...
#&gt;   [5897]     11504        7150
#&gt;   [5898]     11505        7151
#&gt;   [5899]     11506        7152
#&gt;   [5900]     11508        7153
#&gt;   [5901]     11512        7154
#&gt;   -------
#&gt;   queryLength: 11512 / subjectLength: 7154</code></pre>
<pre class="r"><code>summit_distance&lt;- distance(TEAD4_summit[queryHits(hits)], TAZ_summit[subjectHits(hits)])

table(summit_distance)</code></pre>
<pre><code>#&gt; summit_distance
#&gt;   0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19 
#&gt; 241 145 134 123 149 116 117 122 111 113 109 120 116 111 106 116  88 101  84  93 
#&gt;  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39 
#&gt;  93  75  89  91  91  83  69  72  65  70  62  74  72  56  71  73  67  76  62  59 
#&gt;  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59 
#&gt;  55  65  47  46  47  55  37  50  37  36  36  31  31  38  46  32  30  37  43  36 
#&gt;  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78  79 
#&gt;  18  28  30  29  33  34  20  21  26  25  23  20  21  15  27  28  24  24  18  15 
#&gt;  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99 
#&gt;  24  21  11  20  23  13  16  10  10  10  15  18   7  14  13  18   8  16   7   6 
#&gt; 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119 
#&gt;   8  12   9   4   7  10  11  10   7   8  11   4  10   7   7   7   9   9   3   8 
#&gt; 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137 139 140 
#&gt;   6   6   2   4   4   6   4   8   4   1   6   5   6   4   2   4   7   3   6   2 
#&gt; 141 142 143 144 145 146 147 148 149 150 152 153 154 155 157 158 160 161 162 163 
#&gt;   4   2   2   5   3   3   4   1   2   2   5   2   8   6   3   2   1   2   3   2 
#&gt; 164 165 166 168 169 170 171 172 175 177 178 179 180 181 182 183 184 185 186 187 
#&gt;   5   3   2   4   3   2   3   2   1   1   2   4   2   2   3   3   3   1   2   1 
#&gt; 191 192 194 195 196 197 198 199 202 204 205 206 207 208 211 212 213 214 216 217 
#&gt;   1   3   2   1   2   1   3   2   2   1   1   1   2   1   2   1   1   1   3   1 
#&gt; 218 220 221 222 224 228 230 231 232 233 234 235 236 239 243 245 246 247 249 
#&gt;   1   2   2   1   2   1   1   1   2   1   1   1   2   1   1   1   5   1   1</code></pre>
<pre class="r"><code>TEAD4_summit[queryHits(hits)][summit_distance ==0]</code></pre>
<pre><code>#&gt; GRanges object with 241 ranges and 2 metadata columns:
#&gt;         seqnames    ranges strand |             name     score
#&gt;            &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |      &lt;character&gt; &lt;numeric&gt;
#&gt;     [1]     chr1   8137853      * |    TEAD4_peak_29   15.8382
#&gt;     [2]     chr1  17128622      * |    TEAD4_peak_91   30.2536
#&gt;     [3]     chr1  26351760      * |   TEAD4_peak_141   28.7604
#&gt;     [4]     chr1  27538961      * |   TEAD4_peak_154   35.3297
#&gt;     [5]     chr1  31576032      * |   TEAD4_peak_180   15.1534
#&gt;     ...      ...       ...    ... .              ...       ...
#&gt;   [237]     chr9 124659491      * | TEAD4_peak_11273   20.8842
#&gt;   [238]     chrX 101208246      * | TEAD4_peak_11442   10.7734
#&gt;   [239]     chrX 120120430      * | TEAD4_peak_11469   31.8595
#&gt;   [240]     chrX 120734965      * | TEAD4_peak_11472   22.8128
#&gt;   [241]     chrX 152896547      * | TEAD4_peak_11501   26.4184
#&gt;   -------
#&gt;   seqinfo: 24 sequences from an unspecified genome; no seqlengths</code></pre>
<pre class="r"><code>TAZ_summit[subjectHits(hits)][summit_distance ==0]</code></pre>
<pre><code>#&gt; GRanges object with 241 ranges and 2 metadata columns:
#&gt;         seqnames    ranges strand |           name     score
#&gt;            &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;character&gt; &lt;numeric&gt;
#&gt;     [1]     chr1   8137853      * |    TAZ_peak_28   18.2561
#&gt;     [2]     chr1  17128622      * |    TAZ_peak_83   24.3522
#&gt;     [3]     chr1  26351761      * |   TAZ_peak_125   34.6508
#&gt;     [4]     chr1  27538962      * |   TAZ_peak_139   22.6101
#&gt;     [5]     chr1  31576031      * |   TAZ_peak_166   34.7791
#&gt;     ...      ...       ...    ... .            ...       ...
#&gt;   [237]     chr9 124659490      * | TAZ_peak_10496   8.61708
#&gt;   [238]     chrX 101208245      * | TAZ_peak_10652   6.40776
#&gt;   [239]     chrX 120120431      * | TAZ_peak_10678  20.78260
#&gt;   [240]     chrX 120734964      * | TAZ_peak_10683  10.32590
#&gt;   [241]     chrX 152896547      * | TAZ_peak_10711  13.43570
#&gt;   -------
#&gt;   seqinfo: 28 sequences from an unspecified genome; no seqlengths</code></pre>
<p>The built-in <code>distance</code> function returns the pair-wise
distance in absolute value.</p>
<p>Let’s revise it to return negative values when TEAD4 summit precede
the TAZ summit and positive values when TEAD4 summit follows TAZ
summit.</p>
<pre class="r"><code># Compute signed distances
signed_distance &lt;- function(A, B) {
  # Compute unsigned distance
  dist &lt;- distance(A, B)
  
  # Determine signs based on whether A precedes or follows B
  sign &lt;- ifelse(start(A) &lt; start(B), -1, 1)
  
  # Apply sign to distance
  dist * sign
}</code></pre>
<pre class="r"><code>library(dplyr)
library(ggplot2)
summit_distance&lt;- signed_distance(TEAD4_summit[queryHits(hits)],
                                  TAZ_summit[subjectHits(hits)])

distance_df&lt;- table(summit_distance) %&gt;%
  tibble::as_tibble() 

distance_df</code></pre>
<pre><code>#&gt; # A tibble: 383 × 2
#&gt;    summit_distance     n
#&gt;    &lt;chr&gt;           &lt;int&gt;
#&gt;  1 -249                1
#&gt;  2 -246                2
#&gt;  3 -233                1
#&gt;  4 -232                2
#&gt;  5 -231                1
#&gt;  6 -224                2
#&gt;  7 -222                1
#&gt;  8 -221                1
#&gt;  9 -220                1
#&gt; 10 -216                1
#&gt; # ℹ 373 more rows</code></pre>
<p>Let’s plot!!</p>
<pre class="r"><code>distance_df %&gt;%
  ggplot(aes(x=summit_distance, y = n)) +
  geom_point()</code></pre>
<p><img src="figure/04_figure1_a_b_c.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/crazyhottommy/reproduce_genomics_paper_figures/blob/c9a4ca2d11835c56a2b7685b05e4a683ab50bbed/docs/figure/04_figure1_a_b_c.Rmd/unnamed-chunk-13-1.png" target="_blank">c9a4ca2</a>
</td>
<td>
crazyhottommy
</td>
<td>
2024-12-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Hmm, something is off… the summit distance on the x-axis needs to be
reordered</p>
<pre class="r"><code>distance_df %&gt;%
  mutate(summit_distance = as.numeric(summit_distance)) %&gt;%
  arrange(summit_distance) %&gt;%
  ggplot(aes(x=summit_distance, y = n)) +
  geom_point()</code></pre>
<p><img src="figure/04_figure1_a_b_c.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/crazyhottommy/reproduce_genomics_paper_figures/blob/c9a4ca2d11835c56a2b7685b05e4a683ab50bbed/docs/figure/04_figure1_a_b_c.Rmd/unnamed-chunk-14-1.png" target="_blank">c9a4ca2</a>
</td>
<td>
crazyhottommy
</td>
<td>
2024-12-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Let’s connect the points with a line:</p>
<pre class="r"><code>distance_df %&gt;%
  mutate(summit_distance = as.numeric(summit_distance)) %&gt;%
  arrange(summit_distance) %&gt;%
  ggplot(aes(x=summit_distance, y = n)) +
  geom_line()</code></pre>
<p><img src="figure/04_figure1_a_b_c.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/crazyhottommy/reproduce_genomics_paper_figures/blob/c9a4ca2d11835c56a2b7685b05e4a683ab50bbed/docs/figure/04_figure1_a_b_c.Rmd/unnamed-chunk-15-1.png" target="_blank">c9a4ca2</a>
</td>
<td>
crazyhottommy
</td>
<td>
2024-12-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The plot looks too wigglely. Let’s smooth it by average the number of
peaks per 5 bp bin.</p>
<pre class="r"><code>df_binned &lt;- distance_df %&gt;%
  mutate(summit_distance = as.numeric(summit_distance)) %&gt;%
  arrange(summit_distance) %&gt;%
  mutate(bin = floor(summit_distance / 5) * 5) %&gt;%  # Create bins by grouping every 5 bp
  group_by(bin) %&gt;%
  summarise(n = mean(n, na.rm = TRUE))  # Calculate average &#39;n&#39; for each bin</code></pre>
<ul>
<li><code>mutate(bin = floor(summit_distance / 5) * 5)</code>:</li>
</ul>
<p>Groups summit_distance values into 5 bp bins by dividing by 5, taking
the floor (rounding down), and multiplying back by 5 to get the bin
lower bound.</p>
<pre class="r"><code># View the binned dataframe
print(df_binned)</code></pre>
<pre><code>#&gt; # A tibble: 97 × 2
#&gt;      bin     n
#&gt;    &lt;dbl&gt; &lt;dbl&gt;
#&gt;  1  -250  1.5 
#&gt;  2  -235  1.33
#&gt;  3  -225  1.33
#&gt;  4  -220  1   
#&gt;  5  -215  1   
#&gt;  6  -210  2   
#&gt;  7  -205  1.5 
#&gt;  8  -200  1.33
#&gt;  9  -195  1   
#&gt; 10  -190  1   
#&gt; # ℹ 87 more rows</code></pre>
<pre class="r"><code>df_binned %&gt;%
  ggplot(aes(x=bin, y = n)) +
  geom_line() +
  scale_x_continuous(breaks = c(-250, 0, 250)) +
  xlab(&quot;distance to the summit \nof TAZ peaks (bp)&quot;) +
  ylab(&quot;peak density&quot;) +
  theme_classic(base_size = 14) </code></pre>
<p><img src="figure/04_figure1_a_b_c.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/crazyhottommy/reproduce_genomics_paper_figures/blob/c9a4ca2d11835c56a2b7685b05e4a683ab50bbed/docs/figure/04_figure1_a_b_c.Rmd/unnamed-chunk-17-1.png" target="_blank">c9a4ca2</a>
</td>
<td>
crazyhottommy
</td>
<td>
2024-12-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Tip</strong>: how do I what packages to use? google “plotting
genome tracks bioconductor” or ask ChatGPT!!</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>#&gt; R version 4.4.1 (2024-06-14)
#&gt; Platform: aarch64-apple-darwin20
#&gt; Running under: macOS Sonoma 14.1
#&gt; 
#&gt; Matrix products: default
#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0
#&gt; 
#&gt; locale:
#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
#&gt; 
#&gt; time zone: America/New_York
#&gt; tzcode source: internal
#&gt; 
#&gt; attached base packages:
#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  
#&gt; [8] base     
#&gt; 
#&gt; other attached packages:
#&gt;  [1] ggplot2_3.5.1         dplyr_1.1.4           Vennerable_3.1.0.9000
#&gt;  [4] here_1.0.1            rtracklayer_1.64.0    GenomicRanges_1.56.1 
#&gt;  [7] GenomeInfoDb_1.40.1   IRanges_2.38.1        S4Vectors_0.42.1     
#&gt; [10] BiocGenerics_0.50.0   workflowr_1.7.1      
#&gt; 
#&gt; loaded via a namespace (and not attached):
#&gt;  [1] tidyselect_1.2.1            farver_2.1.2               
#&gt;  [3] Biostrings_2.72.1           bitops_1.0-8               
#&gt;  [5] fastmap_1.2.0               RCurl_1.98-1.16            
#&gt;  [7] GenomicAlignments_1.40.0    promises_1.3.0             
#&gt;  [9] XML_3.99-0.17               digest_0.6.36              
#&gt; [11] lifecycle_1.0.4             processx_3.8.4             
#&gt; [13] magrittr_2.0.3              compiler_4.4.1             
#&gt; [15] rlang_1.1.4                 sass_0.4.9                 
#&gt; [17] tools_4.4.1                 utf8_1.2.4                 
#&gt; [19] yaml_2.3.10                 knitr_1.48                 
#&gt; [21] labeling_0.4.3              S4Arrays_1.4.1             
#&gt; [23] curl_5.2.1                  DelayedArray_0.30.1        
#&gt; [25] plyr_1.8.9                  RColorBrewer_1.1-3         
#&gt; [27] abind_1.4-5                 BiocParallel_1.38.0        
#&gt; [29] withr_3.0.0                 grid_4.4.1                 
#&gt; [31] fansi_1.0.6                 git2r_0.35.0               
#&gt; [33] colorspace_2.1-1            scales_1.3.0               
#&gt; [35] SummarizedExperiment_1.34.0 cli_3.6.3                  
#&gt; [37] rmarkdown_2.27              crayon_1.5.3               
#&gt; [39] generics_0.1.3              rstudioapi_0.16.0          
#&gt; [41] httr_1.4.7                  reshape2_1.4.4             
#&gt; [43] rjson_0.2.22                cachem_1.1.0               
#&gt; [45] stringr_1.5.1               zlibbioc_1.50.0            
#&gt; [47] parallel_4.4.1              XVector_0.44.0             
#&gt; [49] restfulr_0.0.15             matrixStats_1.3.0          
#&gt; [51] vctrs_0.6.5                 Matrix_1.7-0               
#&gt; [53] jsonlite_1.8.8              callr_3.7.6                
#&gt; [55] RBGL_1.80.0                 jquerylib_0.1.4            
#&gt; [57] glue_1.7.0                  codetools_0.2-20           
#&gt; [59] ps_1.7.7                    stringi_1.8.4              
#&gt; [61] gtable_0.3.5                later_1.3.2                
#&gt; [63] BiocIO_1.14.0               UCSC.utils_1.0.0           
#&gt; [65] munsell_0.5.1               tibble_3.2.1               
#&gt; [67] pillar_1.9.0                htmltools_0.5.8.1          
#&gt; [69] graph_1.82.0                GenomeInfoDbData_1.2.12    
#&gt; [71] R6_2.5.1                    rprojroot_2.0.4            
#&gt; [73] evaluate_0.24.0             Biobase_2.64.0             
#&gt; [75] lattice_0.22-6              highr_0.11                 
#&gt; [77] Rsamtools_2.20.0            httpuv_1.6.15              
#&gt; [79] bslib_0.8.0                 Rcpp_1.0.13                
#&gt; [81] SparseArray_1.4.8           whisker_0.4.1              
#&gt; [83] xfun_0.46                   fs_1.6.4                   
#&gt; [85] MatrixGenerics_1.16.0       getPass_0.2-4              
#&gt; [87] pkgconfig_2.0.3</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
