<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2024-12-28" />

<title>reproduce figure 1 g,h</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">FigRemake with Chatomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/crazyhottommy/reproduce_genomics_paper_figures">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">reproduce figure 1 g,h</h1>
<h4 class="date">2024-12-28</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-01-04
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>reproduce_genomics_paper_figures/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20241226code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20241226)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20241226code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20241226)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomcrazyhottommyreproducegenomicspaperfigurestree883cb56f69344e885d1adaa6c40f0d551bed1a9etargetblank883cb56a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/crazyhottommy/reproduce_genomics_paper_figures/tree/883cb56f69344e885d1adaa6c40f0d551bed1a9e" target="_blank">883cb56</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomcrazyhottommyreproducegenomicspaperfigurestree883cb56f69344e885d1adaa6c40f0d551bed1a9etargetblank883cb56a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/crazyhottommy/reproduce_genomics_paper_figures/tree/883cb56f69344e885d1adaa6c40f0d551bed1a9e" target="_blank">883cb56</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    data/fastq/
    Ignored:    data/public_data/
    Ignored:    data/reference/

Unstaged changes:
    Modified:   .gitignore
    Modified:   analysis/01_download_fastq_from_GEO.Rmd
    Modified:   analysis/02_align_to_hg38.Rmd
    Modified:   analysis/03_generate_bigwig.Rmd
    Modified:   analysis/04_figure1_a_b_c.Rmd
    Modified:   analysis/05_figure1_d_f.Rmd
    Modified:   analysis/06_figure1_g_h.Rmd
    Modified:   analysis/07_figure1_i_j_k.Rmd
    Modified:   analysis/08_homework.Rmd
    Modified:   analysis/_site.yml
    Modified:   analysis/about.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/06_figure1_g_h.Rmd</code>) and
HTML (<code>docs/06_figure1_g_h.html</code>) files. If you’ve configured
a remote Git repository (see <code>?wflow_git_remote</code>), click on
the hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/crazyhottommy/reproduce_genomics_paper_figures/blob/c9a4ca2d11835c56a2b7685b05e4a683ab50bbed/analysis/06_figure1_g_h.Rmd" target="_blank">c9a4ca2</a>
</td>
<td>
crazyhottommy
</td>
<td>
2024-12-31
</td>
<td>
version 0.1
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/crazyhottommy/reproduce_genomics_paper_figures/c9a4ca2d11835c56a2b7685b05e4a683ab50bbed/docs/06_figure1_g_h.html" target="_blank">c9a4ca2</a>
</td>
<td>
crazyhottommy
</td>
<td>
2024-12-31
</td>
<td>
version 0.1
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Let’s re-create figure 1 g,h!</p>
<p><img src="imgs/figgh.png" /></p>
<div id="figure-1g" class="section level3">
<h3>Figure 1g</h3>
<p>Figure 1g is an <a href="https://igv.org/"><code>IGV</code></a> track
view. We already generated the bigwig files and we can easily load them
to IGV and take a screenshot.</p>
<p>Below is one of the known YAP1 target gene CCN2:</p>
<p><img src="imgs/ccn2.png" /></p>
<p>The paper also shows H3K4me1, H3K4me3 and H3K27ac tracks which the
author analyzed from a <a
href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE49651">previous
dataset</a> for MDA-MB-231 breast cancer cell line.</p>
<p>I will leave the analysis from the raw fastq files as an exercise for
you.</p>
</div>
<div id="plot-the-tracks-from-scratch" class="section level3">
<h3>plot the tracks from scratch</h3>
<p>If you look carefully, the genome track view for the bigwig files is
just a histogram.</p>
<p>If you want to plot the bigwig signal tracks:</p>
<ul>
<li><p>take a look at <a
href="https://phanstiellab.github.io/plotgardener/articles/guides/plotting_multiomic_data.html">plotgardner</a>
bioconductor package.</p></li>
<li><p><a
href="https://bioconductor.org/packages/release/bioc/html/karyoploteR.html">karyoploteR</a>
I used for my <a
href="https://github.com/crazyhottommy/scATACutils">scATACutils
package</a> for plotting <a
href="https://rpubs.com/crazyhottommy/scATAC_tracks">scATACseq
tracks</a>.</p></li>
<li><p><a
href="https://bioconductor.org/packages/devel/bioc/vignettes/Gviz/inst/doc/Gviz.html">Gviz</a></p></li>
<li><p><a
href="https://www.bioconductor.org/packages/release/bioc/html/ggbio.html">ggbio</a></p></li>
<li><p><a
href="https://github.com/deeptools/pyGenomeTracks">pyGenometracks</a> in
<code>python</code>.</p></li>
</ul>
</div>
<div id="figure-1h" class="section level3">
<h3>Figure 1h</h3>
<p>Figure 1h is a pie chart showing the proportion of the annotation of
the YAP/TAZ/TEAD4 common peaks. The annotations are: promoters, active
enhancers, inactive enhancers and not classified.</p>
<ul>
<li>promoters are YAP/TAZ/TEAD4 peaks overlapping with H3K4me3
peaks</li>
<li>active enhancers are YAP/TAZ/TEAD4 peaks overlapping with H3K4me1
and H3K27ac peaks</li>
<li>inactive enhancers are YAP/TAZ/TEAD4 peaks overlapping with H3K4me1
but not H3K37ac peaks</li>
</ul>
<p>Read this <a
href="https://www.cell.com/cell/fulltext/S0092-8674(07)00600-9">classic
ChIP-seq paper</a> from Keji Zhao’s group to understand what genomic
features those histone modifications are associated with.</p>
<p>To do this, I will download the H3K4me1, H3K4me3 and H3K27ac peaks
from <a
href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE49651"
class="uri">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE49651</a>.</p>
<p><img src="imgs/GEO.png" /> click the <code>ftp</code> or
<code>http</code> in the link above, and you will download the
<code>peaks.txt.gz</code> files.</p>
<p>Note, the data were aligned to hg19 human genome. We did our
alignment using hg38 genome. We will need to lift-over the hg19 peak
files to hg38 version.</p>
<pre class="bash"><code>ls data/public_data

GSE49651_MDAMB231.H3K27Ac.hg19.all.MACS.1e-5_peaks.txt.gz GSE49651_MDAMB231.H3K4me3.hg19.ALL.MACS.1e-5_peaks.txt.gz
GSE49651_MDAMB231.H3K4me1.hg19.all.MACS.1e-5_peaks.txt.gz
</code></pre>
<p>Download the chain file from UCSC <a
href="https://hgdownload.soe.ucsc.edu/goldenpath/hg19/liftOver/"
class="uri">https://hgdownload.soe.ucsc.edu/goldenpath/hg19/liftOver/</a>
and the command line tool from <a
href="https://hgdownload.soe.ucsc.edu/admin/exe/"
class="uri">https://hgdownload.soe.ucsc.edu/admin/exe/</a></p>
<pre class="bash"><code>curl -O https://hgdownload.soe.ucsc.edu/goldenpath/hg19/liftOver/hg19ToHg38.over.chain.gz

# I am using a macs with arm64 system
curl -O https://hgdownload.soe.ucsc.edu/admin/exe/macOSX.arm64/liftOver

# make it executable
chmod u+x liftOver</code></pre>
<p>Now, we are ready to lift-over the coordinates from hg19 to hg38:</p>
<pre class="bash"><code>./liftOver GSE49651_MDAMB231.H3K27Ac.hg19.all.MACS.1e-5_peaks.txt.gz hg19ToHg38.over.chain.gz H3K27ac.bed unmapped1.bed

./liftOver GSE49651_MDAMB231.H3K4me3.hg19.ALL.MACS.1e-5_peaks.txt.gz hg19ToHg38.over.chain.gz H3K4me3.bed unmapped2.bed

./liftOver GSE49651_MDAMB231.H3K4me1.hg19.all.MACS.1e-5_peaks.txt.gz hg19ToHg38.over.chain.gz H3K4me1.bed unmapped3.bed

# how many peaks after the liftover
 wc -l *bed
   37480 H3K27ac.bed
   26765 H3K4me1.bed
   17991 H3K4me3.bed
      30 unmapped1.bed
      56 unmapped2.bed
      12 unmapped3.bed</code></pre>
<p>Again, we can use <code>bedtools</code> to do all those intersections
and annotate the YAP1/TAZ/TEAD4 peaks. I will keep everything in R
instead:</p>
<pre class="r"><code>library(rtracklayer) # for reading in bed file
library(here)
library(dplyr)
library(ggplot2)

TAZ_peaks&lt;- import(here(&quot;data/fastq/TAZ_peak/TAZ_peaks.narrowPeak&quot;))
YAP_peaks&lt;- import(here(&quot;data/fastq/YAP_peak/YAP_peaks.narrowPeak&quot;))
TEAD4_peak&lt;- import(here(&quot;data/fastq/TEAD4_peak/TEAD4_peaks.narrowPeak&quot;))

YAP_overlap_TAZ_peaks&lt;- subsetByOverlaps(YAP_peaks, TAZ_peaks)

YAP_overlap_TAZ_peaks_overlap_TEAD4&lt;- subsetByOverlaps(YAP_overlap_TAZ_peaks, TEAD4_peak)
YAP_overlap_TAZ_peaks_overlap_TEAD4</code></pre>
<pre><code>#&gt; GRanges object with 5965 ranges and 6 metadata columns:
#&gt;          seqnames              ranges strand |          name     score
#&gt;             &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |   &lt;character&gt; &lt;numeric&gt;
#&gt;      [1]     chr1     1024628-1025059      * |    YAP_peak_3       494
#&gt;      [2]     chr1     1264837-1265155      * |    YAP_peak_4       148
#&gt;      [3]     chr1     1265320-1265695      * |    YAP_peak_5       131
#&gt;      [4]     chr1     1360618-1360955      * |    YAP_peak_6       306
#&gt;      [5]     chr1     1659298-1659586      * |    YAP_peak_8        45
#&gt;      ...      ...                 ...    ... .           ...       ...
#&gt;   [5961]     chrX 154368850-154369243      * | YAP_peak_9801        90
#&gt;   [5962]     chrX 154596614-154596846      * | YAP_peak_9802       120
#&gt;   [5963]     chrX 154600351-154600918      * | YAP_peak_9803       131
#&gt;   [5964]     chrX 154732680-154732891      * | YAP_peak_9804        58
#&gt;   [5965]     chrX 155888248-155888487      * | YAP_peak_9806       108
#&gt;          signalValue    pValue    qValue      peak
#&gt;            &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;integer&gt;
#&gt;      [1]    16.37320   54.3883  49.46070       192
#&gt;      [2]     8.81841   18.8472  14.89750       188
#&gt;      [3]     8.62413   17.0083  13.14010       112
#&gt;      [4]    14.37360   35.1297  30.64260       148
#&gt;      [5]     5.06808    7.8940   4.58709       102
#&gt;      ...         ...       ...       ...       ...
#&gt;   [5961]     6.13997  12.64820   9.00333       237
#&gt;   [5962]     8.21346  15.84150  12.02720       128
#&gt;   [5963]     8.62413  17.00830  13.14010       417
#&gt;   [5964]     5.74942   9.32421   5.89513        78
#&gt;   [5965]     6.72473  14.58900  10.84320       151
#&gt;   -------
#&gt;   seqinfo: 27 sequences from an unspecified genome; no seqlengths</code></pre>
<p>read in the histone modification peaks:</p>
<pre class="r"><code>H3K4me1&lt;- import(here(&quot;data/public_data/H3K4me1.bed&quot;))
H3K4me3&lt;- import(here(&quot;data/public_data/H3K4me3.bed&quot;))
H3K27ac&lt;- import(here(&quot;data/public_data/H3K27ac.bed&quot;))</code></pre>
<p>From the method section:</p>
<blockquote>
<p>The presence of H3K4me1 and H3K4me3 peaks, their genomic locations
and their overlap were the criteria used to define promoters and
enhancers: i) H3K4me3 peaks not overlapping with H3K4me1 peaks and close
to a TSS (± 5kb) were defined as promoters, as NA otherwise; ii) H3K4me1
peaks not overlapping with H3K4me3 peaks were defined as enhancers; iii)
regions with the co-presence of H3K4me1 and H3K4me3 peaks were visually
inspected on IGV and were defined as promoters, enhancers or NA after
the evaluation of the proximity to a TSS and the comparison of the
enrichment signals. Finally, promoters or enhancers were defined as
active if overlapping with H3K27ac peaks.</p>
</blockquote>
<p>Define (in)active enhancers:</p>
<pre class="r"><code>active_enhancers&lt;- subsetByOverlaps(H3K4me1, H3K27ac)
inactive_enhancers&lt;- subsetByOverlaps(H3K4me1, H3K27ac, invert=TRUE)

promoters&lt;- subsetByOverlaps(H3K4me3, H3K4me1, invert=TRUE)</code></pre>
<blockquote>
<p>YAP/TAZ/TEAD peaks annotation YAP/TAZ/TEAD peaks were annotated as
promoters or enhancers if their summit was overlapping with promoter or
enhancer regions as defined above. Peaks with the summit falling in
regions with no H3K4me1 or H3K4me3 peaks, or in NA regions were defined
as “not assigned” and discarded from subsequent analyses.</p>
</blockquote>
<pre class="r"><code>n_active_enhancers&lt;- subsetByOverlaps(YAP_overlap_TAZ_peaks_overlap_TEAD4,
                                      active_enhancers) %&gt;%
  length()

n_inactive_enhancers&lt;- subsetByOverlaps(YAP_overlap_TAZ_peaks_overlap_TEAD4,
                                        inactive_enhancers) %&gt;%
  length()

n_promoters&lt;- subsetByOverlaps(YAP_overlap_TAZ_peaks_overlap_TEAD4, 
                               promoters) %&gt;%
  length()

n_unclassified&lt;- length(YAP_overlap_TAZ_peaks_overlap_TEAD4) - n_active_enhancers -
  n_inactive_enhancers - n_promoters</code></pre>
<p>Put the numbers in a dataframe:</p>
<pre class="r"><code>annotation_df&lt;- data.frame(category = c(&quot;active_enhancers&quot;, &quot;inactive_enhancers&quot;,
                        &quot;promoters&quot;, &quot;unclassified&quot;),
peak_number = c(n_active_enhancers, n_inactive_enhancers, 
                n_promoters, n_unclassified))


annotation_df</code></pre>
<pre><code>#&gt;             category peak_number
#&gt; 1   active_enhancers        3972
#&gt; 2 inactive_enhancers         118
#&gt; 3          promoters         274
#&gt; 4       unclassified        1601</code></pre>
<p>Let’s plot a pie chart!</p>
<pre class="r"><code>library(ggplot2)

ggplot(annotation_df, aes(x = &quot;&quot;, y = peak_number, fill = category)) +
  geom_bar(stat = &quot;identity&quot;, width = 1) +
  coord_polar(&quot;y&quot;, start = 0) +
  theme_void() + # Remove unnecessary axes
  labs(title = &quot;YAP/TAZ/TEAD4 peaks&quot;) +
  scale_fill_brewer(palette = &quot;Set3&quot;) </code></pre>
<p><img src="figure/06_figure1_g_h.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/crazyhottommy/reproduce_genomics_paper_figures/blob/c9a4ca2d11835c56a2b7685b05e4a683ab50bbed/docs/figure/06_figure1_g_h.Rmd/unnamed-chunk-10-1.png" target="_blank">c9a4ca2</a>
</td>
<td>
crazyhottommy
</td>
<td>
2024-12-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>change the order of the categories by changing the factor level</p>
<pre class="r"><code>annotation_df$category&lt;- factor(annotation_df$category, 
                                levels = c(&quot;promoters&quot;, &quot;active_enhancers&quot;,
                                           &quot;inactive_enhancers&quot;, &quot;unclassified&quot;))

colors&lt;- c(&quot;#8D1E0F&quot;, &quot;#F57D2B&quot;, &quot;#FADAC4&quot;, &quot;#D4DADA&quot;)

ggplot(annotation_df, aes(x = &quot;&quot;, y = peak_number, fill = category)) +
  geom_bar(stat = &quot;identity&quot;, width = 1) +
  coord_polar(&quot;y&quot;, start = 0) +
  theme_void() + # Remove unnecessary axes
  labs(title = &quot;YAP/TAZ/TEAD4 peaks&quot;) +
  scale_fill_manual(values = colors)</code></pre>
<p><img src="figure/06_figure1_g_h.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/crazyhottommy/reproduce_genomics_paper_figures/blob/c9a4ca2d11835c56a2b7685b05e4a683ab50bbed/docs/figure/06_figure1_g_h.Rmd/unnamed-chunk-11-1.png" target="_blank">c9a4ca2</a>
</td>
<td>
crazyhottommy
</td>
<td>
2024-12-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Note, the authors re-called the peaks using their own IgG sample:</p>
<blockquote>
<p>Peak calls and read density tracks were generated using SPP version
1.1148 with default parameters and using as control sample the IgG
ChIP-seq data generated in our laboratory because of the low sequencing
depth of the Input DNA contained in SRP028597.</p>
</blockquote>
<p>This can cause drastically different number of H3K4me1/3 and H3K27ac
peaks. It is not surprising to me that we now have a lot of unclassified
peaks.</p>
<div id="add-the-percentage-to-the-pie-chart" class="section level4">
<h4>Add the percentage to the pie chart</h4>
<pre class="r"><code># Calculate percentages and cumulative positions for labeling
annotation_df &lt;- annotation_df %&gt;%
  dplyr::mutate(
    percentage = peak_number / sum(peak_number) * 100,
    label = paste0(round(percentage, 1), &quot;%&quot;)
  )

annotation_df</code></pre>
<pre><code>#&gt;             category peak_number percentage label
#&gt; 1   active_enhancers        3972  66.588433 66.6%
#&gt; 2 inactive_enhancers         118   1.978206    2%
#&gt; 3          promoters         274   4.593462  4.6%
#&gt; 4       unclassified        1601  26.839899 26.8%</code></pre>
<pre class="r"><code># Create the pie chart
ggplot(annotation_df, aes(x = &quot;&quot;, y = peak_number, fill = category)) +
  geom_bar(stat = &quot;identity&quot;, width = 1) + 
  coord_polar(&quot;y&quot;, start = 0) +
  theme_void() + # Remove unnecessary axes
  labs(title = &quot;YAP/TAZ/TEAD4 peaks&quot;) +
  scale_fill_manual(values = colors) +
  geom_text(aes(label = label), position = position_stack(vjust = 0.5)) # Add percentage labels</code></pre>
<p><img src="figure/06_figure1_g_h.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/crazyhottommy/reproduce_genomics_paper_figures/blob/c9a4ca2d11835c56a2b7685b05e4a683ab50bbed/docs/figure/06_figure1_g_h.Rmd/unnamed-chunk-13-1.png" target="_blank">c9a4ca2</a>
</td>
<td>
crazyhottommy
</td>
<td>
2024-12-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Although we have a bigger proportion of unclassified peaks, the
conclusion is the same: A large proportion of the YAP1/TAZ/TEAD4 peaks
are located at active enhancers.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>#&gt; R version 4.4.1 (2024-06-14)
#&gt; Platform: aarch64-apple-darwin20
#&gt; Running under: macOS Sonoma 14.1
#&gt; 
#&gt; Matrix products: default
#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0
#&gt; 
#&gt; locale:
#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
#&gt; 
#&gt; time zone: America/New_York
#&gt; tzcode source: internal
#&gt; 
#&gt; attached base packages:
#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  
#&gt; [8] base     
#&gt; 
#&gt; other attached packages:
#&gt;  [1] ggplot2_3.5.1        dplyr_1.1.4          here_1.0.1          
#&gt;  [4] rtracklayer_1.64.0   GenomicRanges_1.56.1 GenomeInfoDb_1.40.1 
#&gt;  [7] IRanges_2.38.1       S4Vectors_0.42.1     BiocGenerics_0.50.0 
#&gt; [10] workflowr_1.7.1     
#&gt; 
#&gt; loaded via a namespace (and not attached):
#&gt;  [1] tidyselect_1.2.1            farver_2.1.2               
#&gt;  [3] Biostrings_2.72.1           bitops_1.0-8               
#&gt;  [5] fastmap_1.2.0               RCurl_1.98-1.16            
#&gt;  [7] GenomicAlignments_1.40.0    promises_1.3.0             
#&gt;  [9] XML_3.99-0.17               digest_0.6.36              
#&gt; [11] lifecycle_1.0.4             processx_3.8.4             
#&gt; [13] magrittr_2.0.3              compiler_4.4.1             
#&gt; [15] rlang_1.1.4                 sass_0.4.9                 
#&gt; [17] tools_4.4.1                 utf8_1.2.4                 
#&gt; [19] yaml_2.3.10                 knitr_1.48                 
#&gt; [21] S4Arrays_1.4.1              labeling_0.4.3             
#&gt; [23] curl_5.2.1                  DelayedArray_0.30.1        
#&gt; [25] RColorBrewer_1.1-3          abind_1.4-5                
#&gt; [27] BiocParallel_1.38.0         withr_3.0.0                
#&gt; [29] grid_4.4.1                  fansi_1.0.6                
#&gt; [31] git2r_0.35.0                colorspace_2.1-1           
#&gt; [33] scales_1.3.0                SummarizedExperiment_1.34.0
#&gt; [35] cli_3.6.3                   rmarkdown_2.27             
#&gt; [37] crayon_1.5.3                generics_0.1.3             
#&gt; [39] rstudioapi_0.16.0           httr_1.4.7                 
#&gt; [41] rjson_0.2.22                cachem_1.1.0               
#&gt; [43] stringr_1.5.1               zlibbioc_1.50.0            
#&gt; [45] parallel_4.4.1              XVector_0.44.0             
#&gt; [47] restfulr_0.0.15             matrixStats_1.3.0          
#&gt; [49] vctrs_0.6.5                 Matrix_1.7-0               
#&gt; [51] jsonlite_1.8.8              callr_3.7.6                
#&gt; [53] jquerylib_0.1.4             glue_1.7.0                 
#&gt; [55] codetools_0.2-20            ps_1.7.7                   
#&gt; [57] stringi_1.8.4               gtable_0.3.5               
#&gt; [59] later_1.3.2                 BiocIO_1.14.0              
#&gt; [61] UCSC.utils_1.0.0            munsell_0.5.1              
#&gt; [63] tibble_3.2.1                pillar_1.9.0               
#&gt; [65] htmltools_0.5.8.1           GenomeInfoDbData_1.2.12    
#&gt; [67] R6_2.5.1                    rprojroot_2.0.4            
#&gt; [69] evaluate_0.24.0             Biobase_2.64.0             
#&gt; [71] lattice_0.22-6              highr_0.11                 
#&gt; [73] Rsamtools_2.20.0            httpuv_1.6.15              
#&gt; [75] bslib_0.8.0                 Rcpp_1.0.13                
#&gt; [77] SparseArray_1.4.8           whisker_0.4.1              
#&gt; [79] xfun_0.46                   fs_1.6.4                   
#&gt; [81] MatrixGenerics_1.16.0       getPass_0.2-4              
#&gt; [83] pkgconfig_2.0.3</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
